### ISSUE TITLE
Authenticating as the current user to a HTTP proxy

### ISSUE THREAD

**Issue by TrevorPilley** [sshnet/SSH.NET#250]
I can see in the ``ConnectionInfo`` I can specify a username and password (as strings), however if I want to use the credentials of the user running an application which uses SSH.NET to authenticate against a proxy server I don't want to have to capture those details somehow. Is it possible to authenticate as the current user (e.g. ``CredentialsCache.DefaultNetworkCredentials``)?

**Comment by CosmKiwi** [comment]
I agree. Storing and sending domain credentials to SshNet.ConnectionInfo to authenticate to a proxy server is not ideal.
